submitter/submission.py

```diff
@@ -25,7 +24,8 @@ def __init__(
         self,
         attributes: dict,
         result_queue: str,
-        result_message: dict | None = None,
+        *,
+        result_message: dict | str | None = None,
```

Comment:
Forgive my ignorance, what is the function of `*` here?

Reply:
No forgiveness needed!  I was not familiar with this syntax until just a couple of months ago.

By including the `*` after the positional arguments (no defaults) it then _requires_ that all following arguments are keyword arguments (and will not work positionally).

I noticed that this class [gets instantiated in two ways](https://github.com/MITLibraries/dspace-submission-service/blob/33603e1921148b064e418c6edd8788939e9dc6a2/submitter/submission.py#L81-L94), what I thought of as "invalid" and "valid" messages.  Because the arguments are so different, and I think even kind of messed with the order, it felt like a nice time to use this and _require_ that anything but the required `attributes` and `result_queue` are passed as keyword arguments.

I don't think this syntax is necessary very often, but this felt like a good time for it.  More information in [PEP 3102 – Keyword-Only Arguments](https://peps.python.org/pep-3102/).

Reply:
Got it, PEP context very much appreciated! A good solution for this scenario